src/main/java/stepsdesign/BeforeActions.java

```diff
@@ -32,11 +32,16 @@
 public class BeforeActions {
   public static Scenario scenario;
   public static File myObj;
+  public static boolean beforeAllFlag = true;
 
   @Before(order = 0)
   public void setUp(Scenario scenario) throws IOException {
+    if (beforeAllFlag) {
```

Comment:
why do we need this flag? Is setUp called multiple times?

Reply:
does it need to be volatile?

Reply:
Yes Cucumber `@Before` `setUp()` gets executed before each scenario.
Cucumber does not have `@BeforeAll `and `@Afterall` tags which will execute before first and after last scenario. We want to setup selenium driver before first scenario and teardown after final scenario. So flag is needed.